src/gram_matrix.jl

```diff
@@ -134,18 +142,42 @@ end
 function MP.polynomial(p::GramMatrix{T,B,U}) where {T,B,U}
     return MP.polynomial(p, U)
 end
-function MP.polynomial(p::GramMatrix, ::Type{S}) where {S}
-    return MP.polynomial(value_matrix(p), p.basis, S)
+
+function MA.operate!(op::SA.UnsafeAddMul{typeof(*)}, p::SA.AlgebraElement, w::SA.AlgebraElement, g::GramMatrix)
```

Comment:
@kalmarek This is the new way to compute `p - sum w_i g_i`. Actually, since the basis if `p` is already determined and is a `SA.ExplicitBasis`, this can potentially be completely allocation-free :rocket: